numpy/core/arrayprint.py

```diff
@@ -521,6 +509,8 @@ def array2string(a, max_line_width=None, precision=None,
     if formatter is None:
         formatter = _formatter
 
+    if style is not None and a.shape == () and a.dtype.names is None:
+        return style(a[()])
```

Comment:
But note that when it goes there, you can't use `PyArray_ToScalar`, as that would circumvent array subclasses that override `[()]` (like maskedarray)